lib/linter/node-event-generator.js

```diff
@@ -185,12 +185,24 @@ function compareSpecificity(selectorA, selectorB) {
 /**
  * Parses a raw selector string, and throws a useful error if parsing fails.
  * @param {string} rawSelector A raw AST selector
+ * @param {Set<string>} knownTypes The known node types
  * @returns {Object} An object (from esquery) describing the matching behavior of this selector
  * @throws {Error} An error if the selector is invalid
  */
-function tryParseSelector(rawSelector) {
+function tryParseSelector(rawSelector, knownTypes) {
+
+    const rawSelectorWithoutExit = rawSelector.replace(/:exit$/u, "");
+
+    // no need to parse known types
+    if (knownTypes.has(rawSelectorWithoutExit)) {
+        return {
+            type: "identifier",
+            value: rawSelectorWithoutExit
+        };
+    }
```

Comment:
With the ESLint configuration for our codebase (enabled rules have 195 different selectors, out of which 113 are identifiers), this change improves performance by a couple of ms. I'm getting different results for the same run, but it's like ~16ms instead of ~18ms in total spent for running this function. Since we are [caching](https://github.com/eslint/eslint/blob/ec2d8307850dd039e118c001416606e1e0342bc8/lib/linter/node-event-generator.js#L210-L226) parsed selectors, this improvement doesn't scale with the number of files being linted, so it looks like it won't bring much.